sdk-common/src/main/java/dev/restate/sdk/annotation/Shared.java

```diff
@@ -13,6 +13,10 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * Defines a method as a Shared handler. It can be used only on methods of either {@link
+ * VirtualObject} or {@link Workflow}. This implies the annotation {@link Handler}.
```

Comment:
Add something like:
`Shared handlers can execute concurrently with the other handlers of Virtual Objects or Workflows. They can therefore not set or clear state.`